packages/p2p/src/components/composite-calendar/composite-calendar.tsx

```diff
@@ -0,0 +1,210 @@
+};
+
+const TwoMonthPickerLoadable = Loadable({
+    loader: () => import(/* webpackChunkName: "two-month-picker" */ './two-month-picker'),
+    loading: () => null,
+    render(loaded, props) {
+        const Component = loaded.default;
+        return <Component {...props} />;
+    },
+});
+
+type TDateFieldComponentProps = {
+    label: string;
+    placeholder: string;
+    showCalendar: (e: string) => void;
+    value: string;
+};
+
+const DateFieldComponent = ({ label, placeholder, showCalendar, value }: TDateFieldComponentProps) => {
+    const { ui } = useStore();
+    const { current_focus, setCurrentFocus } = ui;
+
+    return (
+        <InputField
+            current_focus={current_focus}
+            data_testid={`dt_calendar_input_${label}`}
+            icon={CalendarIcon}
+            is_read_only
+            name={`${label}_date`}
+            onClick={() => showCalendar(label)}
+            placeholder={placeholder}
+            setCurrentFocus={setCurrentFocus}
+            value={value}
+        />
+    );
+};
+
+const CompositeCalendar = (props: TCompositeCalendarProps) => {
+    const { ui } = useStore();
+    const { current_focus, setCurrentFocus } = ui;
+    const { from, to, onChange } = props;
+
+    const [show_to, setShowTo] = React.useState(false);
+    const [show_from, setShowFrom] = React.useState(false);
+    const [days_duration_list] = React.useState([
+        {
+            value: 'all_time',
+            label: localize('All time'),
+            onClick: () => selectDateRange(),
+            duration: 0,
+        },
+        {
+            value: 'today',
+            label: localize('Today'),
+            onClick: () => selectDateRange(1),
+            duration: 1,
+        },
+        {
+            value: 'last_7_days',
+            label: localize('Last 7 days'),
+            onClick: () => selectDateRange(7),
+            duration: 7,
+        },
+        {
+            value: 'last_30_days',
+            label: localize('Last 30 days'),
+            onClick: () => selectDateRange(30),
+            duration: 30,
+        },
+        {
+            value: 'last_60_days',
+            label: localize('Last 60 days'),
+            onClick: () => selectDateRange(60),
+            duration: 60,
+        },
+        {
+            value: 'last_quarter',
+            label: localize('Last quarter'),
+            onClick: () => selectDateRange(90),
+            duration: 90,
+        },
+    ]);
+
+    const wrapper_ref = React.useRef<HTMLInputElement>(null);
+
+    const validateClickOutside = (event: MouseEvent) => !wrapper_ref.current?.contains(event.target as Node);
+
+    const selectDateRange = (new_from?: number) => {
+        hideCalendar();
+        onChange({
+            from: new_from ? toMoment().startOf('day').subtract(new_from, 'day').add(1, 's') : undefined,
+            to: toMoment().endOf('day'),
+            is_batch: true,
+        });
+    };
+
+    const getToDateLabel = () => {
```

Comment:
could you add tsdocs to these as well, lets try to slowly add more documentation for others in the future to understand 🙏